pyfstat/utils/sft.py

```diff
@@ -175,3 +179,163 @@ def get_official_sft_filename(
+        SFTConstraint.maxStartTime = None
+    else:
+        SFTConstraint.minStartTime = lal.LIGOTimeGPS(tstart)
+        SFTConstraint.maxStartTime = SFTConstraint.minStartTime + duration
+
+    SFTConstraint.timestamps = None  # FIXME: not currently supported
+    if timestamps is not None:
+        raise NotImplementedError("Timestamps not yet supported in this function.")
+
+    logger.info(
+        "SFT Constraints: [minStartTime:{}, maxStartTime:{}]".format(
+            SFTConstraint.minStartTime,
+            SFTConstraint.maxStartTime,
+        )
+    )
+
+    return SFTConstraint
+
+
+def plot_spectrogram(
+    sftfilepattern: str,
+    detector: str,
+    savefig: Optional[bool] = False,
+    outdir: Optional[str] = ".",
+    label: Optional[str] = None,
+    quantity: Optional[str] = "power",
+    sqrtSX: Optional[float] = None,
+    fMin: Optional[float] = None,
+    fMax: Optional[float] = None,
+    constraints: Optional[lalpulsar.SFTConstraints] = None,
+    **kwargs,
+) -> matplotlib.axes.Axes:
+    """
+    Compute spectrograms of a set of SFTs.
+    This is useful to produce visualizations of the Doppler modulation of a CW signal.
+
+    Parameters
+    ----------
+    sftfilepattern:
+        Pattern to match SFTs using wildcards (`*?`) and ranges [0-9];
+        multiple patterns can be given separated by colons.
+    detector:
+        Name of the detector of the data that will be plot.
+    savefig:
+        If True, save the figure in `outdir`.
+        If False, return an axis object without saving to disk.
+    outdir:
+        Output folder.
+    label:
+        Output filename.
+    quantity:
+        Magnitude to be plotted.
+        It can be "power" for power, "normpower" for normalized power, "real" for the real part of
+        the SFTs, and "imag" for the imaginary part of the SFTs.
+        Set to "power" by default.
+    sqrtSX:
+        Amplitude spectral density of the data.
+        Only needed if `quantity = "normpower".
+    fMin, fMax:
+        Restrict frequency range to `[fMin, fMax]`.
+        If None, retrieve the full frequency range.
+    constraints:
+        Constraints to be fed into XLALSFTdataFind to specify detector,
+        GPS time range or timestamps to be retrieved.
+    kwarg: dict
+        Other kwargs, only used to be passed to `matplotlib.pcolormesh`.
+
+    Returns
+    -------
+    ax: matplotlib.axes.Axes
+        The axes object containing the plot.
+    """
+
+    logger.info("Loading SFT data")
+    frequency, timestamps, fourier_data = get_sft_as_arrays(
+        sftfilepattern, fMin, fMax, constraints
+    )
+
+    if savefig:
+        if label is None:  # pragma: no cover
+            raise ValueError("Label needed to save the figure")
+        else:
+            plotfile = os.path.join(outdir, label + ".png")
+
+    plt.rcParams["axes.grid"] = False  # turn off the gridlines
+    fig, ax = plt.subplots(figsize=(0.8 * 16, 0.8 * 9))  # FIXME: make configurable
```

Comment:
Let's do this before merging - you can add a `figsize` argument with tuple type and `(16,9)` as default value - I don't think the 0.8 factor here had any deeper meaning.